lib/index.js

```diff
@@ -52,6 +52,7 @@ internals.schema = Joi.object({
     password: Joi.string().required(),
     clientId: Joi.string().required(),
     clientSecret: Joi.string().required().allow(''),
+    signatureMethod: Joi.string().valid('HMAC-SHA1', 'RSA-SHA1').when('provider.protocol', { is: 'oauth', otherwise: Joi.forbidden() }),
```

Comment:
I don't think there is a reason for a user of a provider would need to choose this option? If there is not, then it should be included in the provider object just above.

Also, when the protocol is oauth, do `then: Joi.default('HMAC-SHA1')` . This will ensure backwards compatibility and ensure it's always set. I know you take care of it  on line 382 of lib/oauth.js but why not let Joi handle it for you.